src-tauri/src/commands/item.rs

```diff
@@ -0,0 +1,77 @@
+use rusqlite::params;
+use tauri::{command, State};
+
+use crate::{
+    errors::CommandResult,
+    filters::{get_filtered_items, Filter},
+    schema::Item,
+    AppState,
+};
+
+/// Wraps a basic `SELECT * FROM items` query.
+///
+/// # Examples
+/// ```javascript
+///     let items: Item[] = [];
+///     async function get_items() {
+///         items = await invoke("get_items");
+///     };
+/// ```
+///
+#[command]
+#[allow(clippy::needless_pass_by_value)]
+pub fn get_items(state: State<AppState>, filter: Option<Filter>) -> CommandResult<Vec<Item>> {
+    let db = state.db.conn.lock().unwrap();
+    let mut items = Vec::new();
+    if let Some(filter) = filter {
```

Comment:
Instead of conditionally running `get_filtered_items` we could always run the `get_filtered_items` as a step (`apply_filters`?) and from that function just return unchanged data if there were no filters passed.  That way the filter function could handle its own logic.

Reply:
Good idea, that logic can live in `filters.rs` as well, keeping `commands/items.rs` clean of any implementation details